File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int,int> mpp;
        mpp[0]=1;
        int res=0;
        int diff=0;
        int curSum=0;

        for(auto n : nums){
            curSum+=n;
            diff = curSum-k;
            if(mpp.find(diff)!=mpp.end())
            {
                res+=mpp[diff];
            }
            mpp[curSum]++;
        }
        return res;
        
    }
};

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        hm = defaultdict(int)
        hm[0] = 1
        res = 0 
        curSum = 0 
        diff = 0 
        for num in nums:
            curSum+=num
            diff = curSum-k
            if diff in hm:
                res+=hm[diff]
            hm[curSum]+=1
        return res
        